lib/Cake/Controller/Component/FlashComponent.php

```diff
@@ -82,12 +82,18 @@ public function set($message, $options = array()) {
 		}
 		$options['element'] = $plugin . 'Flash/' . $element;
 
-		CakeSession::write('Message.' . $options['key'], array(
+		$messages = CakeSession::read('Message.' . $options['key']);
+
+		$newMessage = array(
```

Comment:
We also have `clear` as a class level configuration option which makes it easy to preserve existing behavior through a configuration change.